artiq/gui/entries.py

```diff
@@ -99,9 +99,17 @@ def state_to_value(state):
     def default_state(procdesc):
         if "default" in procdesc:
             return procdesc["default"]
-        elif "min" in procdesc and procdesc["min"] is not None:
-            return procdesc["min"]
         else:
+            have_max = "max" in procdesc and procdesc["max"] is not None
+            have_min = "min" in procdesc and procdesc["min"] is not None
+            if have_max and have_min and procdesc["min"] <= 0 < procdesc["max"]:
+                return 0
+            elif have_min and not have_max:
+                if procdesc["min"] > 0:
```

Comment:
``min`` is inclusive (and ``max`` is exclusive). It doesn't change the behavior but ``>=`` would be clearer here.

Reply:
Other than that it looks good to me.